src/person/controllers/authenticate_controller.py

```diff
@@ -0,0 +1,118 @@
+import app.settings as api_settings
+from app.utils import failure_response
+from app.utils import success_response
+from django.contrib.auth import authenticate
+from django.contrib.auth import login
+from django.contrib.auth.models import User
+from django.utils import timezone
+from google.auth.transport import requests
+from google.oauth2 import id_token
+from person.models import Person
+from rest_framework import status
+from rest_framework.authtoken.models import Token
+
+
+class AuthenticateController:
+    def __init__(self, request, data, serializer):
+        self._request = request
+        self._data = data
+        self._serializer = serializer
+
+    def authenticate(self, username, password):
+        user = authenticate(username=username, password=password)
+        if user is not None:
+            login(self._request, user)
+            return user, status.HTTP_200_OK
+        else:
+            return None, status.HTTP_403_FORBIDDEN
+
+    def create_access_token(self, user):
+        """Creates or retrieves current access token for user. If expired, then create new access token."""
+        token, _ = Token.objects.get_or_create(user=user)
+        elapsed_seconds = (timezone.now() - token.created).total_seconds()
+        if api_settings.ACCESS_TOKEN_AGE <= elapsed_seconds:
+            token.delete()
+            token = Token.objects.create(user=user)
+        return token.key
+
+    def create_user(self, user_data):
+        """Creates new user (Django auth) from user_data"""
+        return User.objects._create_user(**user_data)
+
+    def create_person(self, person_data):
+        """Creates new Person object from person_data"""
+        person = Person(**person_data)
+        person.save()
+        return person
+
+    def create_token_info(self, token_info):
+        """Create token info"""
+        username = token_info.get("email")
+        google_user_id = token_info.get("sub")
+        password = api_settings.AUTH_PASSWORD_SALT + google_user_id
+        first_name = token_info.get("given_name")
+        last_name = token_info.get("family_name")
+        netid = token_info.get("email").split("@")[0]
```

Comment:
this isn't guaranteed to be right though.. like are we ensuring that emails are valid? someone could be sending you something erroneous.. maybe we should be more careful?